### PR TITLE
show Xur only when available

### LINKED ISSUE #6506: Xûr shows up in vendor response when he's not actually around
Not sure what we want to do about this but the API now returns Xur as a vendor even when he's not available.

### PR DESCRIPTION
**Pull request by delphiactual**
Closes #6506

### CODE REVIEW

**Comment by nev-r** [comment]
i think this is only gating the location, not gating vendor from displaying

**Comment by delphiactual** [comment]
you are absolutely correct, i will close this while i look into this more, while xur is gone!

**Review by nev-r — COMMENTED** [review]
image
i think? this will hide eva & saladin on tuesday-thursday

File: src/app/vendors/Vendors.tsx [hunk 1]
```diff
@@ -182,6 +184,29 @@ function Vendors({
     vendorGroups = filterVendorGroupsToSearch(vendorGroups, searchQuery, filterItems);
   }
 
+  if (
+    currentXur().start === undefined &&
+    vendorGroups &&
+    Object.values(vendorGroups).some((v) => v.def.hash === VENDOR_GROUPS.LIMITED_TIME)
```
Comment (nev-r) [inline src/app/vendors/Vendors.tsx:190]: vendorGroups is already an array as far as i can tell, shouldn't need object.values

File: src/app/vendors/Vendors.tsx [hunk 2]
```diff
@@ -182,6 +184,29 @@ function Vendors({
...
+  ) {
+    const vgIndex = vendorGroups
+      .map(function (v) {
+        return v.def.hash;
+      })
+      .indexOf(VENDOR_GROUPS.LIMITED_TIME);
+    if (Object.values(vendorGroups[vgIndex].vendors).some((v) => v.def.hash === VENDORS.XUR)) {
```
Comment (nev-r) [inline src/app/vendors/Vendors.tsx:197]: same here. array not object

File: src/app/vendors/Vendors.tsx [hunk 3]
```diff
@@ -182,6 +184,29 @@ function Vendors({
     vendorGroups = filterVendorGroupsToSearch(vendorGroups, searchQuery, filterItems);
```
Comment (nev-r) [inline src/app/vendors/Vendors.tsx:184]: @bhollis can you confirm why there's two `vendorGroups &&` (L180, L183)?
is there an undocumented possibility of undefinedness?

**Comment by delphiactual** [comment]
| Current | With PR|
|---------|----------|
| image |  image |

**Review by nev-r — APPROVED** [review]

**Pull request merged**